packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -48,6 +48,47 @@ const String _kVMSnapshotData = 'vm_snapshot_data';
 const String _kIsolateSnapshotData = 'isolate_snapshot_data';
 const String _kIsolateSnapshotInstr = 'isolate_snapshot_instr';
 
+/// Allows to mock `buildBundle`.
+class BundleFactory {
+  Future<void> buildBundle({
+    TargetPlatform platform,
+    BuildMode buildMode,
+    String mainPath = defaultMainPath,
+    String manifestPath = defaultManifestPath,
+    String applicationKernelFilePath,
+    String depfilePath,
+    String privateKeyPath = defaultPrivateKeyPath,
+    String assetDirPath,
+    String packagesPath,
+    bool precompiledSnapshot = false,
+    bool reportLicensedPackages = false,
+    bool trackWidgetCreation = false,
+    String compilationTraceFilePath,
+    List<String> extraFrontEndOptions = const <String>[],
+    List<String> extraGenSnapshotOptions = const <String>[],
+    List<String> fileSystemRoots,
+    String fileSystemScheme,
+  }) => build(
```

Comment:
Not sure if this will help make things cleaner, but can you try making `build` an instance method of `BundleFactory`. Then I guess `buildBundle` would be the constructor, and you would use it like `BundleFactory(...)..build();` Might help avoid repeating this long parameter list so many times.

Reply:
I renamed  `BundleFactory` as `BundleBuilder`.  I also merged `BundleBuilder().build(..)` and `build(..)`.   I thought about moving the parameters to the constructor, but that would make the initialization in `build_bundle.dart` a little complex. e.g.  can't be done in the constructor.  Also, we use a similar pattern in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/attach.dart#L276-L286